### PR TITLE
Add iroh-one to CI steps

### PR DESCRIPTION
**Pull request by fabricedesre**
- build with the 'uds-gateway' feature to catch build errors.
- push the iroh-one binary to S3.

Ping @Arqu

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -348,7 +350,7 @@ jobs:
           profile-name: default
       - run:
           name: build x86_64 release
-          command: cargo build --release -v
+          command: cargo build --release -v --features=uds-gateway
```
Comment (Arqu) [inline .circleci/config.yml:353]: Need to replicate this for `build_release_aarch64` & `build_release_osx`

**Review by Arqu — CHANGES_REQUESTED** [review]
Looks good, lets just make sure other architectures are supported too.

File: .circleci/config.yml (hunk 1 above)
Comment (fabricedesre) [reply .circleci/config.yml:353]: Done in 0ec0ec0c97362647768b980f547597cbdf84542b

**Review by Arqu — CHANGES_REQUESTED** [review]
I'm sorry but can we take off the `--features=uds-gateway` from the builds. Don't think I want to have those uploaded as the "default" builds. Those who want to include specific features should generally build their own and if it becomes widely used we can move that into a config value rather than compile time thing.

**Comment by fabricedesre** [comment]
> I'm sorry, been thinking about this, but can we take off the `--features=uds-gateway` from the builds. Don't think I want to have those uploaded as the "default" builds. Those who want to include specific features should generally build their own and if it becomes widely used we can move that into a config value rather than compile time thing.

I understand your argument, but that also means that we have no CI coverage for build breaking with that feature enabled. Or should we build both with and without the feature, and only upload the featureless one?

**Comment by Arqu** [comment]
> > I'm sorry, been thinking about this, but can we take off the `--features=uds-gateway` from the builds. Don't think I want to have those uploaded as the "default" builds. Those who want to include specific features should generally build their own and if it becomes widely used we can move that into a config value rather than compile time thing.
> 
> I understand your argument, but that also means that we have no CI coverage for build breaking with that feature enabled. Or should we build both with and without the feature, and only upload the featureless one?

Right, that makes sense. So what I would do instead is introduce a new CI job and run `cargo check --all-features` and `cargo check --no-default-features` in it as a prequel to building releases and make it depend on `cargo fetch` and `clippy` and `rustfmt`. That way we get build coverage. Potentially extend the linux tests to run tests in the same fashion.

**Comment by Arqu** [comment]
This is now covered by CI. I'd be happy to revisit adding iroh one to the official builds.

**Comment by dignifiedquire** [comment]
Closing  this PR. @fabricedesre @Arqu whoever gets to it first, lets get iroh-one added to the binary list.

**Pull request closed without merging**